src/java/org/apache/cassandra/metrics/ClientMetrics.java

```diff
@@ -51,20 +84,62 @@ public final class ClientMetrics
     private Meter protocolException;
     private Meter unknownException;
 
+    private static String AUTH_SUCCESS = "AuthSuccess";
+
+    private static String AUTH_FAILURE = "AuthFailure";
+
+    private static String CONNECTED_NATIVE_CLIENTS = "ConnectedNativeClients";
+
     private ClientMetrics()
     {
     }
 
+    /**
+     * @deprecated by {@link #markAuthSuccess(String)}
+     */
+    @Deprecated(since="5.1", forRemoval = true)
     public void markAuthSuccess()
     {
-        authSuccess.mark();
+        markAuthSuccess(null);
+    }
+
+    public void markAuthSuccess(final String mode)
+    {
+        markAuthMeter(authSuccess, AUTH_SUCCESS, mode);
     }
 
+    /**
+     * @deprecated by {@link #markAuthFailure(String)}
+     */
+    @Deprecated(since="5.1", forRemoval = true)
     public void markAuthFailure()
     {
         authFailure.mark();
     }
 
+    public void markAuthFailure(final String mode)
+    {
+        markAuthMeter(authFailure, AUTH_FAILURE, mode);
+    }
+
+    private void markAuthMeter(final Meter meter, final String meterPrefix, final String mode)
+    {
+        meter.mark();
+
+        if (mode != null)
+        {
+            Map<String, Meter> meterMap = meterPrefix.equals(AUTH_SUCCESS) ? authSuccessByMode : authFailureByMode;
```

Comment:
@tolbertam  btw why cant we pass the map in the parameter? you know which map you want based on what method you call it from so you do not need to know prefix at all, no?

Reply:
yeah that would make more sense to just pass in the Map of meters instead of resolving it based on the passed in value, will do :+1: